File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.java
import java.util.*;
class Solution {
    public int[] solution(String myString) {
        List<Integer> li = new ArrayList<>();
        int cnt = 0;
        for(int i=0; i<myString.length(); i++){
            if (myString.charAt(i) == 'x') {
                li.add(cnt);
                cnt = 0;
            }
            else cnt++;
        }
        li.add(cnt);
        int[] answer = new int[li.size()];
        for(int i=0; i<li.size(); i++) {
            answer[i] = li.get(i);
        }
        return answer;
    }
}

File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string myString) {
    vector<int> answer;
    int cnt = 0;
    for(int i=0; i<myString.size(); i++) {
        if (myString[i] == 'x') {
            answer.push_back(cnt);
            cnt = 0;
            if (i == myString.size()-1) {
                answer.push_back(cnt);
            }
        }
        else cnt++;
    }
    if (cnt) answer.push_back(cnt);
    return answer;
}